### PR TITLE
Update jira team to be `applications`

### PR DESCRIPTION
**Pull request by tvoran**
Updates the jira sync team from `foundations` => `applications`.

### CODE REVIEW

**Comment by tomhjp** [comment]
While we're editing it anyway, maybe could update it to use the common workflow? Similar to https://github.com/hashicorp/vault-plugin-database-snowflake/blob/main/.github/workflows/jira.yaml

**Comment by tvoran** [comment]
@tomhjp Oh yes, that looks much better! Done in 1489f7a.

File: .github/workflows/jira.yaml [hunk 1]
```diff
@@ -1,72 +1,19 @@
...
-
-    - name: Reopen ticket
-      if: github.event.action == 'reopened' && steps.search.outputs.issue
-      uses: atlassian/gajira-transition@v2.0.1
-      with:
-        issue: ${{ steps.search.outputs.issue }}
-        transition: "Pending Triage"
+    uses: hashicorp/vault-workflows-common/.github/workflows/jira.yaml@main
+    # assuming you use Vault to get secrets
+    # if you use GitHub secrets, use secrets.XYZ instead of steps.secrets.outputs.XYZ
```
Comment (tomhjp) [inline .github/workflows/jira.yaml:13]: Optional nit: 

```suggestion
```

**Review by tomhjp — APPROVED** [review]
LGTM 👍

**Pull request merged**